kernel/src/actions/visitors.rs

```diff
@@ -268,6 +268,45 @@ impl DataVisitor for RemoveVisitor {
     }
 }
 
+#[derive(Default)]
```

Comment:
i'll probably remove this visitor, but perhaps we want to see about giving connectors a public way to extract the descriptor from their data if it's more convenient